pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -94,6 +112,7 @@ func ApplyFeatureGates() {
 		// Register the priority function to specific provider too.
 		factory.InsertPriorityKeyToAlgorithmProviderMap(factory.RegisterPriorityFunction2(priorities.ResourceLimitsPriority, priorities.ResourceLimitsPriorityMap, nil, 1))
 	}
+	return featureDependency
```

Comment:
`featureDependency` is sort of a stable thing. I do believe we should keep it as somewhere as (const) variable, rather than calculating it everytime in `ApplyFeatureGates()` - remember that `ApplyFeatureGates()` may be called in integration tests, and we don't want to do the calculation multiple times.

If we follow this idea, `ApplyFeatureGates()` should read from `featureDependency` and register/unregister predKey/predFunc and prioKey/prioFunc properly.